fibonacci.js

```diff
@@ -0,0 +1,158 @@
+function runLocalFibonacci() {
+  count = 0;
+  fSequence = [0, 1];
+  fIndex = document.getElementById("fInput").value;
+  document.getElementById("displayF").innerText = fibonacci(fIndex);
+  document.getElementById("fInput").value = " ";
+}
+
+// document.getElementById("fButton").addEventListener("click", runLocalFibonacci)
+
+function runLocalFibonacci(fIndex) {
+  let fSequence = [0, 1];
+  for (let i = 2; i <= fIndex; i++) {
+    fSequence.push(fSequence[i - 2] + fSequence[i - 1]);
+  }
+  finalResult.classList.remove(
+    "spinner-border",
+    "text-primary",
+    "font-weight-bold"
+  );
+  finalResult.classList.add("fs-4", "text-decoration-underline");
+  finalResult.innerText = fSequence[fSequence.length - 1];
+}
+
+let button = document.getElementById("fButton");
+let input = document.getElementById("fInput");
+let finalResult = document.getElementById("displayF");
+let error = document.querySelector(".error");
+let serverError = document.getElementById("serverError");
+let resultContainer = document.querySelector(".resultContainer");
+let spinnerResult = document.getElementById("spinnerResult");
+let saveCalculation = document.getElementById("saveCalculation");
+let select = document.getElementById("select")
+let showSelectedValue = document.querySelector(".text-muted")
+
+button.addEventListener("click", chooseLocalOrServer);
+input.addEventListener("focus", clearInput);
+
+function chooseLocalOrServer() {
+    if (input.value.length) {
+        button.disabled = true;
+   
+    if (input.value > 50) {
+        input.classList.add("border-danger", "text-danger");
+        error.classList.remove("d-none");
+      } 
+  else if (saveCalculation.checked) {
+    runServerFibonacci();
+  } else  if (input.value<=0){
+    finalResult.classList.remove(
+        "spinner-border",
+        "text-primary",
+        "font-weight-bold"
+      );
+      finalResult.classList.add("text-danger");
+      finalResult.innerText = "Error: number can't be smaller than 1";
+    } else {
+    runLocalFibonacci(input.value);
+  }
+}}
+
+function runServerFibonacci() {
+      finalResult.classList.add("spinner-border", "text-primary");
+      fetch(`http://localhost:5050/fibonacci/${input.value}`)
+        .then((res) => {
+          if (res.ok) return res.json();
+          else {
+            return res.text().then((text) => {
+              throw new Error(text);
+            });
+          }
+        })
+        .then((data) => {
+          finalResult.classList.remove(
+            "spinner-border",
+            "text-primary",
+            "font-weight-bold"
+          );
+          finalResult.classList.add("fs-4", "text-decoration-underline");
+          finalResult.innerText = data.result;
+          fetchResults(selectedValue);
```

Comment:
since the variable selectedValue is not declered or changed inside runServerFibonacci, there is no need to pass it as a parameter to fetchResults()